pkg/kubelet/network/dns/dns_windows.go

```diff
@@ -64,30 +64,22 @@ var (
 )
 
 func getHostDNSConfig(resolverConfig string) (*runtimeapi.DNSConfig, error) {
-	if resolverConfig != "" && resolverConfig != hostResolvConf {
-		err := fmt.Errorf(`Unexpected resolver config value: "%s". Expected "" or "%s".`, resolverConfig, hostResolvConf)
-		klog.ErrorS(err, "Cannot get host DNS Configuration.")
-		return nil, err
+	if resolverConfig != hostResolvConf {
```

Comment:
Is "Host" sufficiently unambiguous now?  For example, could I point this flag at a file named "File" ?  If so, why would "Host" not mean a file named "Host" ?

Is there some syntax Windows comprehends that is clear not a possible filename?  "Host:" or "host://" or something without the ambiguity?

Or can we assert that this must be an absolute path?  Does a relative path like "File" work today?  We didn't used to assert that, so it's kind of a breaking change.

Reply:
If relative paths **do not** work today, I'm in favor of asserting this must be an absolute path.
@claudiubelu - can you check if relative paths work today? (and determine where the path is relative *from* so we can document that?)

Reply:
@thockin ``Host`` is the name we've chosen. We could instead use ``HostRegistry``, which is going to be less ambiguous, IMO.

@marosset relative paths are relative to the executable. So, if we're to ``os.Open("resolv.conf")`` file, it would open the ``resolv.conf`` file from the same folder as the binary.

I've updated the PR. Now, we're first checking if the given ``resolverConfig`` is a file or not. If it's a file (using ``os.Stat``), then we're treating it as a file. If it's not a file, we're checking if it's ``resolverConfig`` is ``Host``, and if so, we're getting the Host's DNS configuration. If it's an empty string, an empty DNS configuration is returned.